tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -2253,6 +2254,36 @@ LogicalResult ConvertTFLPadOp::matchAndRewrite(
   return success();
 }
 
+LogicalResult ConvertTFLMirrorPadOp::matchAndRewrite(
+    Operation* op, PatternRewriter& rewriter) const {
+  auto tfl_mirrorpad_op = cast<TFL::MirrorPadOp>(op);
+
+  RankedTensorType output_type =
+      tfl_mirrorpad_op.getResult().getType().dyn_cast<RankedTensorType>();
+  if (!output_type) {
+    return rewriter.notifyMatchFailure(op, "output type isn't a ranked tensor");
+  }
+
+  mlir::TFL::MirrorPaddingType padding = tfl_mirrorpad_op.mode();
+  StringRef mode = "UNKNOWN";
+  switch (padding) {
+    case mlir::TFL::MirrorPaddingType::REFLECT:
+      mode = "REFLECT";
+      break;
+    case mlir::TFL::MirrorPaddingType::SYMMETRIC:
+      mode = "SYMMETRIC";
+      break;
+  }
```

Comment:
Could you just return here in unknown case and avoid the notify failure internally? (currently these two are exhaustive and verified to be one of these on the op, so one could even assert/mark unreachable here, but to be conservative as things change this is defensible as these are just strings rather than an enum)

Reply:
Sure thing. Return here in unknown case. Note that the mode passing in `convertMirrorPadCommon` changes to enum from string as discussed.